pyroute2/nftables/main.py

```diff
@@ -125,3 +137,83 @@ def rule(self, cmd, **kwarg):
                 expressions.extend(exp)
             kwarg['expressions'] = expressions
         return self._command(nft_rule_msg, commands, cmd, kwarg)
+
+    def sets(self, cmd, **kwarg):
+        '''
+        Example::
+            nft.sets("add", table="filter", name="test0", key_type="ipv4_addr")
+            nft.sets("get", table="filter", name="test0")
+            nft.sets("del", table="filter", name="test0")
+        '''
+        commands = {
+            'add': NFT_MSG_NEWSET,
+            'get': NFT_MSG_GETSET,
+            'del': NFT_MSG_DELSET,
+        }
+
+        if cmd in 'add':
+            if 'key_len' not in kwarg:
+                key_type, key_len, _ = DATA_TYPE_NAME_TO_INFO.get(kwarg['key_type'])
+                kwarg["key_type"] = key_type
+                kwarg["key_len"] = key_len
+            kwarg['id'] = 1
+
+        return self._command(nft_set_msg, commands, cmd, kwarg)
+
+    def set_elems(self, cmd, **kwarg):
+        '''
+        Example::
+            nft.set_elems("add", table="filter", set="test0", elements=["10.2.3.4", "10.4.3.2"])
+            nft.set_elems("get", table="filter", set="test0")
+            nft.set_elems("del", table="filter", set="test0", elements=["10.2.3.4"])
+        '''
+        commands = {
+            'add': NFT_MSG_NEWSETELEM,
+            'get': NFT_MSG_GETSETELEM,
+            'del': NFT_MSG_DELSETELEM,
+        }
+        set_info = self.sets("get", table=kwarg["table"], name=kwarg["set"])
+        data_type_name = DATA_TYPE_ID_TO_NAME.get(set_info.get_attr("NFTA_SET_KEY_TYPE"))
+        if data_type_name is not None:
+            _, _, modifier = DATA_TYPE_NAME_TO_INFO[data_type_name]
+            modifier = modifier()
+            modifier.header = None
+        else:
+            modifier = None
+
+        if cmd == "get":
+            msg = nft_set_elem_list_msg()
+            msg['attrs'] = [
+                ["NFTA_SET_ELEM_LIST_TABLE", kwarg["table"]],
+                ["NFTA_SET_ELEM_LIST_SET", kwarg["set"]]
+            ]
+            msg = self.request_get(msg, NFT_MSG_GETSETELEM)[0]
+            elements = []
+            for elem in msg.get_attr('NFTA_SET_ELEM_LIST_ELEMENTS'):
+                if modifier is not None:
+                    # Need to find a better way
+                    modifier.data = elem.get_attr('NFTA_SET_ELEM_KEY').get_attr("NFTA_DATA_VALUE")
+                    modifier.length = 4 + len(modifier.data)
+                    modifier.decode()
+                    elements.append(modifier.value)
+            return elements
+
+        data_type_name = DATA_TYPE_ID_TO_NAME.get(set_info.get_attr("NFTA_SET_KEY_TYPE"))
+        if modifier is not None:
+            elements = []
+            for elem in kwarg.pop("elements"):
+                modifier.value = elem
+                modifier.encode()
+                elements.append(modifier["value"])
+        else:
+            kwarg.pop("elements")
+
+        kwarg["elements"] = [
```

Comment:
I already started to work on kwarg filters for `IPRoute`, see `/pyroute2/requests/`; eventually we could make some common API that solves common transformations like `str 255.255.255.0 ⇒ int 24`, or `iterable ( int interface_index ) ⇒ int interface_index`

Reply:
I will see that thanks !